apps/req.c

```diff
@@ -115,13 +116,19 @@ const OPTIONS req_options[] = {
     {"text", OPT_TEXT, '-', "Text form of request"},
     {"x509", OPT_X509, '-',
      "Output an x509 structure instead of a cert request"},
+    {"CA", OPT_CA, '<', "Issuer certificate to use with -x509"},
+    {"CAkey", OPT_CAKEY, 's',
+     "Issuer private key to use with -x509; default is -CA arg"},
```

Comment:
I am really not sure the req command needs to duplicate the functionality of x509 command here. IMO the -x509 option for req command should produce self-signed certificates only.

Reply:
Having the `-CA` and `-CAkey` options not only in the `x509` app but also in the `req` app
makes the two apps more consistent and in particular is very helpful for the test cases I added to `25-test_req.t`.
As I mentioned recently in #13720 there is a large overlap between the `x509`, `req`, and `ca` app,
and the tendency has been to (non always consistently) extend `x509` and `req`,
also with functionality that one would expect for the `ca` app.
Part of the reason must be that `ca` is rather inconvenient to use.